### DISCUSSION TITLE
helm aliases should be resolved in the helm datasource

### DISCUSSION THREAD

**Discussion by scrocquesel** [renovatebot/renovate#13160] [Suggest an Idea]
### What would you like Renovate to be able to do?

When extracting helm dependencies with regex manager, helm alias are not evaluate by the datasource. Currently it is up to the different helm managers to resolve the alias with the global helmAlias and they do it in different ways.
https://github.com/renovatebot/renovate/blob/ede1d2b5872c766fe98125d34680c167b42bc0b5/lib/manager/helmfile/extract.ts#L58-L72, https://github.com/renovatebot/renovate/blob/ede1d2b5872c766fe98125d34680c167b42bc0b5/lib/manager/helmv3/extract.ts#L62-L73, https://github.com/renovatebot/renovate/blob/ede1d2b5872c766fe98125d34680c167b42bc0b5/lib/manager/helm-requirements/extract.ts#L47-L56

This config is not available to regex manager and registryUrlTemplate do not allow to split/switch.

### If you have any ideas on how this should be implemented, please tell us here.

Not at all. I'm not sure the datasource getReleases could have access to the alias config.

It's currently more of a reminder related to some discussions regarding a potential helm managers refactor https://github.com/renovatebot/renovate/issues/7547#issuecomment-716188164 or a use case for https://github.com/renovatebot/renovate/issues/9577

### Is this a feature you are interested in implementing yourself?

Maybe

**Comment by rarkins** [comment]
The presence of duplicated code in managers doesn't necessarily mean it should be moved to the datasource. Aliases is a helm-specific concept we would prefer not to add to our generic datasource interface and `registryUrls` should be used instead.

**Comment by scrocquesel** [comment]
Sure, but in some way, the config is open to manager which is also generic. Why the difference between datasource and manager. 
That said, an issue is that aliases can either be global or local to file being processed. Meaning, the manager should also be able to forward local aliases to the datasource.

But then we are stuck to only use supported helm file format to extract helm dependency. As an example, I have a custom ansible playbook to play helm install. In some part of the configuration, I was able to rewrote the config to read a helmfile (but don't want to use this tool), and in some other I can't. I'm left with a set of regex rules with part of the regex matching the alias to set staticaly set the full registry name with a registryUrlTemplate. The aliases config is now spread out in multiple places in renovate config.

Maybe being able to use the aliases map in the templating engine could help.

**Reply by rarkins** [reply]
Converted this to a discussion. We don’t want to reinvent each package manager concept within regex manager either as it would become too complex and unmanageable.

Based on your description, this requirement is too much of an “edge case” and would add too much complexity that the benefit to you would be outweighed by the complexity downsides for us and other users.
